### ISSUE TITLE
(Monthly) Scheduled transactions do not appear to follow intended schedule

### ISSUE THREAD

**Issue by mutedbytes** [codinguser/gnucash-android#721]
#### Steps to reproduce the behaviour
1. Create monthly scheduled transaction, set to repeat on "same day each month"
2. Create a 2nd monthly schedule transaction, but on a different date

#### Expected behaviour
I would expect the first scheduled transaction to repeat every month on specified date of transaction 1, and the second schedule transaction to occur later the following month on the different date of transaction 2.
Example: 
Transaction 1 =  Aug 1, repeat on Sep 1
Transaction 2 = Aug 16, repeat on Sep 16

#### Actual behaviour
The actual behavior appears to be that all monthly scheduled transactions take place on Sep 1.
It also appears getting the scheduled transactions to actually run is sporadic and inconsistent.

#### Software specifications
* GnuCash Android version: 2.2.1
* System Android version: 6.0.1
* Device type: LG Nexus 5

**Comment by codinguser** [comment]
@rivaldi8 Will your new JobScheduler help here?

**Comment by rivaldi8** [comment]
@codinguser I don't think so. Those changes were mainly for the restrictions of Android 8. I'd say it's more likely something to do with the code deciding when to run the transactions.

**Comment by nikopp** [comment]
I can confirm this bug on
- GnuCash Android version: 2.4.0
- System Android version: 8.0.0
- Device type: Samsung Galaxy A3
